### PR TITLE
Preview media file

### PR DESCRIPTION
**Pull request by madalinazanficu**

### CODE REVIEW

**Comment by madalinazanficu** [comment]
How it looks now:
file:///home/madalina/Downloads/WhatsApp%20Image%202024-05-29%20at%2014.09.19.jpeg

Used resizeMode "contain" with a fixed height, because the uploaded image has fixed dimensions for height and width.

File: mobile/app/(app)/(drawer)/(tabs)/(quick-report)/report-details/[reportId].tsx [hunk 1]
```diff
@@ -103,4 +103,47 @@ const ReportDetails = () => {
...
 };
 
+interface attachementProps {
+  attachment: QuickReportAttachmentAPIResponse;
+}
+
+const MediaPreview = (props: attachementProps) => {
+  const { attachment } = props;
+
+  console.log(attachment);
```
Comment (birloiflorian) [inline mobile/app/(app)/(drawer)/(tabs)/(quick-report)/report-details/[reportId].tsx:113]: Remove logs

File: mobile/components/AudioPlayer.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * TODO: This component would need support for integrating plyaback controls.
+ * For now, it just plays the audio file.
+ * Recommended: react-native track player
+ */
+const AudioPlayer = (props: AudioPlayerProps) => {
+  const { uri } = props;
+  const [sound, setSound] = useState<Audio.Sound | undefined>(undefined);
+  const [status, setStatus] = useState<any>({});
```
Comment (birloiflorian) [inline mobile/components/AudioPlayer.tsx:18]: we should avoid using any. please update with an actual type.

File: mobile/components/AudioPlayer.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    async function loadSound() {
+      const { sound } = await Audio.Sound.createAsync({ uri }, { shouldPlay: false });
+      sound.setOnPlaybackStatusUpdate((status) => setStatus(status));
+      setSound(sound);
+    }
+    loadSound();
+
+    return () => {
+      if (sound) {
+        console.log("Unloading Sound");
```
Comment (birloiflorian) [inline mobile/components/AudioPlayer.tsx:30]: Remove log

File: mobile/components/MediaDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+interface DialogProps extends AlertDialogProps {
+  // what you press on in order to open the dialog
+  trigger?: ReactNode;
+  // dialog header
+  header?: ReactNode;
+  // content inside dialog
+  content?: ReactNode;
+}
+
+const MediaDialog: React.FC<DialogProps> = ({ header, content, trigger, ...props }) => {
```
Comment (birloiflorian) [inline mobile/components/MediaDialog.tsx:13]: I don't understand what is the difference between this and the DialogModal component that we already have?

**Review by birloiflorian — CHANGES_REQUESTED** [review]

File: mobile/package.json [hunk 5]
```diff
@@ -71,7 +73,8 @@
     "react-native-toast-message": "^2.2.0",
     "superjson": "^2.2.1",
     "tamagui": "^1.93.2",
-    "zod": "^3.23.3"
+    "zod": "^3.23.3",
+    "expo-av": "~13.10.6"
```
[the remark this answers is not in the document]
Comment (madalinazanficu) [reply mobile/package.json]: 👍 removed

File: mobile/components/VideoPlayer.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import { Video, ResizeMode } from "expo-av";
+import { View } from "tamagui";
+
+interface VideoPlayerProps {
+  uri: string;
+}
+const VideoPlayer = (props: VideoPlayerProps) => {
+  const { uri } = props;
+  const video = React.useRef<Video>(null);
+  const [_status, setStatus] = React.useState({});
```
[the remark this answers is not in the document]
Comment (madalinazanficu) [reply mobile/components/VideoPlayer.tsx]: 👍 removed

File: mobile/components/VideoPlayer.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+  return (
+    <View>
```
[the remark this answers is not in the document]
Comment (madalinazanficu) [reply mobile/components/VideoPlayer.tsx]: 👍 removed

File: mobile/components/AudioPlayer.tsx (hunk 3 above)
Comment (madalinazanficu) [reply mobile/components/AudioPlayer.tsx]: 👍 done

File: mobile/components/AudioPlayer.tsx (hunk 2 above)
Comment (madalinazanficu) [reply mobile/components/AudioPlayer.tsx]: 👍 done

File: mobile/app/(app)/(drawer)/(tabs)/(quick-report)/report-details/[reportId].tsx (hunk 1 above)
Comment (madalinazanficu) [reply mobile/app/(app)/(drawer)/(tabs)/(quick-report)/report-details/[reportId].tsx]: 👍 done

**Pull request closed without merging**